rama-pac/src/connector/mod.rs

```diff
@@ -0,0 +1,97 @@
+use crate::pac_parser::ProxyDirective;
+use crate::{pac_fetcher::fetch_pac, pac_parser::parse_pac_file};
+use rama_core::{Context, Service, error::BoxError};
+use rama_http::{Uri, service::client::HttpClientExt};
+use rama_http_types::Request;
+
+pub struct PACConnector<S, W> {
+    pub service: S,
+    pub web_client: W,
+    pub pac_uri: Uri,
+}
+
+impl<S, W> PACConnector<S, W>
+where
+    W: HttpClientExt + Send + Sync + 'static,
+{
+    pub fn new(service: S, web_client: W, pac_uri: Uri) -> Self {
+        Self {
+            service,
+            web_client,
+            pac_uri,
+        }
+    }
+}
+
+impl<S, W, Body> Service<Request<Body>> for PACConnector<S, W>
+where
+    S: Service<Request<Body>, Error: Into<BoxError> + Send + Sync + 'static>,
+    W: HttpClientExt + Send + Sync + 'static,
+    Body: Clone + Send + Sync + 'static,
+    W::ExecuteError: std::error::Error + Send + Sync + 'static,
+    W::ExecuteResponse: Into<String>,
+{
+    type Response = S::Response;
+    type Error = BoxError;
+
+    async fn serve(
+        &self,
+        ctx: Context,
+        req: Request<Body>,
+    ) -> Result<Self::Response, Self::Error> {
+        let pac_file = fetch_pac(&self.web_client, &self.pac_uri)
+            .await
+            .unwrap();
```

Comment:
given it's wip this is ok, but obviously in the final code errors like this should be handled or bubbled up. no unwraps please unless you have a very good reason